cli/flox-rust-sdk/src/models/environment/managed_environment.rs

```diff
@@ -450,6 +451,40 @@ impl Environment for ManagedEnvironment {
 
         Ok(())
     }
+
+    fn migrate_to_v1(
+        &mut self,
+        flox: &Flox,
+        migration_info: MigrationInfo,
+    ) -> Result<(), EnvironmentError> {
+        let mut generations = self
+            .generations()
+            .writable(flox.temp_dir.clone())
+            .map_err(ManagedEnvironmentError::CreateFloxmetaDir)?;
+        let mut temporary = generations
+            .get_current_generation()
+            .map_err(ManagedEnvironmentError::CreateGenerationFiles)?;
+
+        let metadata = match (
+            migration_info.needs_manifest_migration,
+            migration_info.needs_upgrade,
+        ) {
+            (true, true) => "Migrated manifest to v1 and upgraded packages",
+            (true, false) => "Migrated manifest to v1", // and locked
+            (false, true) => "Upgraded packages",
+            _ => unreachable!(),
```

Comment:
Suggestion: add a message e.g. `unreachable!("called with invalid migration metadata")` so we can easier debug this if ever encountered

Reply:
Fixing